Linux headless template_release build: use of undeclared identifier 'RenderingServer'

### Tested versions

Godot master branch ab4c5a594ab11cf6446fa819de639bb71de9ccbf

### System information

Gentoo Linux LLVM amd64 musl libc

### Issue description

When compiling Godot with all modules disabled, the build fails with the following error:
```Swift
[  7%] clang++ -o main/performance.linuxbsd.template_release.x86_64.llvm.o -c -std=gnu++17 -fno-exceptions -Wctor-dtor-privacy -Wnon-virtual-dtor -flto=thin -pipe -Os -Wall -Wextra -Wwrite-strings -Wno-unused-parameter -Wshadow-field-in-constructor -Wshadow-uncaptured-local -Wno-ordered-compare-function-pointers -Wimplicit-fallthrough -Werror -DNDEBUG -DNO_EDITOR_SPLASH -DDISABLE_DEPRECATED -DXKB_ENABLED -DJOYDEV_ENABLED -DLINUXBSD_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -D_3D_DISABLED -DADVANCED_GUI_DISABLED -DTHREADS_ENABLED -DCLIPPER2_ENABLED -DZSTD_STATIC_LINKING_ONLY -Ithirdparty/libpng -Ithirdparty/zstd -Ithirdparty/zlib -Ithirdparty/clipper2/include -Iplatform/linuxbsd -I. main/performance.cpp
platform/linuxbsd/os_linuxbsd.cpp:280:6: error: use of undeclared identifier 'RenderingServer'
  280 |         if (RenderingServer::get_singleton() == nullptr) {
      |             ^
platform/linuxbsd/os_linuxbsd.cpp:289:39: error: use of undeclared identifier 'RenderingServer'
  289 |         const String rendering_device_name = RenderingServer::get_singleton()->get_video_adapter_name(); // e.g. `NVIDIA GeForce GTX 970`
      |                                              ^
platform/linuxbsd/os_linuxbsd.cpp:290:41: error: use of undeclared identifier 'RenderingServer'
  290 |         const String rendering_device_vendor = RenderingServer::get_singleton()->get_video_adapter_vendor(); // e.g. `NVIDIA`
      |                                                ^
3 errors generated.
scons: *** [platform/linuxbsd/os_linuxbsd.linuxbsd.template_release.x86_64.llvm.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:26.890]
```

### Steps to reproduce

```bash
scons $(cat options)
````
options:
```bash
platform=linuxbsd
target=template_release
arch=x86_64
dev_build=no
optimize=size
debug_symbols=no
separate_debug_symbols=no
lto=thin
production=yes
threads=yes
deprecated=no
precision=single
minizip=no
brotli=no
xaudio2=no
vulkan=no
opengl3=no
d3d12=no
openxr=no
use_volk=no
disable_exceptions=yes
dev_mode=no
tests=no
fast_unsafe=no
compiledb=no
verbose=yes
progress=yes
warnings=extra
werror=yes
vsproj=no
disable_3d=yes
disable_advanced_gui=yes
modules_enabled_by_default=no
no_editor_splash=yes
use_precise_math_checks=no
scu_build=no
linker=mold
use_llvm=yes
use_static_cpp=no
use_coverage=no
use_ubsan=no
use_asan=no
use_lsan=no
use_tsan=no
use_msan=no
use_sowrap=no
alsa=no
pulseaudio=no
dbus=no
speechd=no
fontconfig=no
udev=no
x11=no
wayland=no
libdecor=no
touch=no
execinfo=no
module_astcenc_enabled=no
module_basis_universal_enabled=no
module_bmp_enabled=no
module_camera_enabled=no
module_csg_enabled=no
module_cvtt_enabled=no
module_dds_enabled=no
module_enet_enabled=no
module_etcpak_enabled=no
module_freetype_enabled=no
module_gdscript_enabled=yes
module_glslang_enabled=no
module_gltf_enabled=no
module_gridmap_enabled=no
module_hdr_enabled=no
module_jpg_enabled=no
module_jsonrpc_enabled=no
module_ktx_enabled=no
module_lightmapper_rd_enabled=no
module_mbedtls_enabled=no
module_meshoptimizer_enabled=no
module_minimp3_enabled=no
minimp3_extra_formats=no
module_mobile_vr_enabled=no
module_mono_enabled=no
module_msdfgen_enabled=no
module_multiplayer_enabled=no
module_navigation_enabled=no
module_noise_enabled=no
module_ogg_enabled=no
module_openxr_enabled=no
module_raycast_enabled=no
module_regex_enabled=no
module_squish_enabled=no
module_svg_enabled=no
module_text_server_adv_enabled=no
graphite=no
module_text_server_fb_enabled=no
module_tga_enabled=no
module_theora_enabled=no
module_tinyexr_enabled=no
module_upnp_enabled=no
module_vhacd_enabled=no
module_vorbis_enabled=no
module_webp_enabled=no
module_webrtc_enabled=no
module_websocket_enabled=no
module_webxr_enabled=no
module_xatlas_unwrap_enabled=no
module_zip_enabled=no
```

### Minimal reproduction project (MRP)

none

--- a/platform/linuxbsd/os_linuxbsd.cpp
+++ b/platform/linuxbsd/os_linuxbsd.cpp
@@ -34,6 +34,7 @@
 #include "core/io/dir_access.h"
 #include "main/main.h"
 #include "servers/display_server.h"
+#include "servers/rendering_server.h"
 
 #ifdef X11_ENABLED
 #include "x11/display_server_x11.h"

